code_challenges/code_wars/football_cards.py

```diff
@@ -0,0 +1,55 @@
+def football_cards(team_cards):
+    a_players = 11
+    b_players = 11
+    a_cards = {}
+    b_cards = {}
+    while a_players > 6 and b_players > 6:
+        for card in team_cards:
+            if a_players == 6 or b_players == 6:
+                break
```

Comment:
```
    while a_players > 6 and b_players > 6:
        for card in team_cards:
            if a_players == 6 or b_players == 6:
                break
```
Do you need this while loop at all?

Reply:
You should read up on `re` the [Python built in](https://docs.python.org/3/library/re.html) for matching and finding things.

An example implementation:
```
import re

team_cards = ["A4R", "A6R", "A8R", "A10R", "A11R"]
for card in team_cards:
    match = re.match(r"([AB])([0-9]+)([YR])", card)
    team, player, penalty = match.groups()
    print(team, player, penalty)
```

While regex/regular expressions can be a lot to think about and write, this is the perfect case were it would make the code a lot more readable.

Reply:
i tried getting rid of the while loop earlier and the code didnt work. which i didnt understand. but i just re did it and it does still work. i must have missed pulling something back when i did it before.